rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1044,68 +1089,66 @@ private void processReleaseRequests() {
-                        ),
-                        exception
-                );
-                return false;
-            }
-
-            if (activations.isActive(ConsensusRule.RSKIP146)) {
-                Keccak256 rskTxHash = releaseRequest.getRskTxHash();
-                // Add the TX
-                releaseTransactionSet.add(generatedTransaction, rskExecutionBlock.getNumber(), rskTxHash);
-                // For a short time period, there could be items in the release request queue that don't have the rskTxHash
-                // (these are releases created right before the consensus rule activation, that weren't processed before its activation)
-                // We shouldn't generate the event for those releases
-                if (rskTxHash != null) {
-                    // Log the Release request
-                    eventLogger.logReleaseBtcRequested(rskTxHash.getBytes(), generatedTransaction, releaseRequest.getAmount());
-                }
-            } else {
-                releaseTransactionSet.add(generatedTransaction, rskExecutionBlock.getNumber());
-            }
+            addPegoutTxToReleaseTransactionSet(generatedTransaction, releaseTransactionSet, releaseRequest);
 
             // Mark UTXOs as spent
+            List<UTXO> selectedUTXOs = result.get().getSelectedUTXOs();
             availableUTXOs.removeAll(selectedUTXOs);
 
             // TODO: (Ariel Mendelzon, 07/12/2017)
             // TODO: Balance adjustment assumes that change output is output with index 1.
             // TODO: This will change if we implement multiple releases per BTC tx, so
             // TODO: it would eventually need to be fixed.
             // Adjust balances in edge cases
-            adjustBalancesIfChangeOutputWasDust(generatedTransaction, releaseRequest.getAmount());
+            adjustBalancesIfChangeOutputWasDust(generatedTransaction, releaseRequest.getAmount(), wallet);
 
             return true;
         });
     }
 
+    private void processReleasesInBatch(ReleaseRequestQueue releaseRequestQueue,
+                                        ReleaseTransactionBuilder txBuilder,
+                                        List<UTXO> availableUTXOs,
+                                        ReleaseTransactionSet releaseTransactionSet,
+                                        Wallet wallet) {
+        long currentBlockNumber = rskExecutionBlock.getNumber();
+        long nextPegoutCreationBlockNumber = getNextPegoutCreationBlockNumber();
+
+        if (currentBlockNumber >= nextPegoutCreationBlockNumber) {
+            // batch pegout transactions
+            Optional<ReleaseTransactionBuilder.BuildResult> result = txBuilder.buildBatchedPegouts(releaseRequestQueue.getEntries());
+
+            if (!result.isPresent()) {
+                logger.warn(
+                    "Couldn't build a release BTC tx for <{}, with sum {}>",
+                    releaseRequestQueue.getEntries().hashCode(),
```

Comment:
not sure if the hash code will be useful tbh

Reply:
Resolved